diff --git a/SConstruct b/SConstruct
index 77cfd13..6a9399e 100644
--- a/SConstruct
+++ b/SConstruct
@@ -676,7 +676,7 @@ if nix:
                          "-Winvalid-pch"] )
     # env.Append( " -Wconversion" ) TODO: this doesn't really work yet
     if linux:
-        env.Append( CCFLAGS=["-Werror", "-pipe"] )
+        env.Append( CCFLAGS=[ "-pipe"] )
         if not has_option('clang'):
             env.Append( CCFLAGS=["-fno-builtin-memcmp"] ) # glibc's memcmp is faster than gcc's
 
