--- a/SConstruct	2012-11-28 16:34:34.000000000 +0400
+++ b/SConstruct	2012-11-28 16:34:49.000000000 +0400
@@ -685,7 +685,7 @@
                          "-Winvalid-pch"] )
     # env.Append( " -Wconversion" ) TODO: this doesn't really work yet
     if linux:
-        env.Append( CCFLAGS=["-Werror", "-pipe"] )
+        env.Append( CCFLAGS=["-pipe"] )
         if not has_option('clang'):
             env.Append( CCFLAGS=["-fno-builtin-memcmp"] ) # glibc's memcmp is faster than gcc's
 
