diff --git a/src/say.m b/src/say.m
index 6d08187..e0012ef 100644
--- a/src/say.m
+++ b/src/say.m
@@ -140,14 +140,7 @@ vsay(int level, const char *filename, int line, const char *error, const char *f
 		if (*f == '/' && *(f + 1) != '\0')
 			filename = f + 1;
 
-	time_t now = (time_t) ev_now();
-	struct tm tm;
-	localtime_r(&now, &tm);
-
-	/* Print time in format 2012-08-07 18:30:00.634 */
-	p += strftime(buf + p, len - p, "%F %H:%M", &tm);
-	p += snprintf(buf + p, len - p, ":%06.3f",
-		      ev_now() - now + tm.tm_sec);
+	p += snprintf(buf + p, len - p, "%.6f",ev_now());
 
 	p += snprintf(buf + p, len - p, " [%i] %i/%s", getpid(),
 		      fiber->fid, fiber->name);